compiler/rustc_session/src/options.rs

```diff
@@ -150,6 +151,9 @@ top_level_options!(
 
         target_triple: TargetTriple [TRACKED],
 
+        /// Effective logical environment used by `env!`/`option_env!` macros
+        logical_env: FxIndexMap<String, String> [TRACKED],
```

Comment:
This shouldn't be marked as tracked. This will cause all env vars to be included in the crate hash even if they aren't actually used according to the dep info. This will cause compilation to be non-reproducible as now all env vars end up in the crate hash even if actually unused.

Reply:
This is not all env vars but just the `--env` args.